includes/restapi/class-notification-controller.php

```diff
@@ -81,4 +91,144 @@ public function get_notifications( $request ) {
+					'type'        => 'string',
+					'format'      => 'date-time',
+					'context'     => array( 'view', 'edit', 'embed' ),
+				),
+				'expires_at'   => array(
+					'description' => __( "The datetime the notification expires, in the site's timezone." ),
+					'type'        => 'string',
+					'format'      => 'date-time',
+					'context'     => array( 'view', 'edit', 'embed' ),
+				),
+				'priority'     => array(
+					'description' => __( 'The priority of the notification.' ),
+					'type'        => 'string',
+					'context'     => array( 'view', 'edit', 'embed' ),
+				),
+				'title'        => array(
+					'description' => __( 'The title of the notification.' ),
+					'type'        => 'string',
+					'context'     => array( 'view', 'edit', 'embed' ),
+				),
+				'message'      => array(
+					'description' => __( 'The message content of the notification.' ),
+					'type'        => 'string',
+					'context'     => array( 'view', 'edit', 'embed' ),
+				),
+				'meta'         => array(
+					'description' => __( 'The metadata of the notification.' ),
+					'type'        => 'string',
+					'context'     => array( 'view', 'edit', 'embed' ),
+				),
+			),
+		);
+
+		// Cache generated schema on endpoint instance.
+		$this->schema = $schema;
+
+		return $this->add_additional_fields_schema( $this->schema );
+	}
+
+	/**
+	 * Retrieves the query params for collections.
+	 *
+	 *
+	 * @return array Comments collection parameters.
+	 */
+	public function get_collection_params() {
+		$query_params = parent::get_collection_params();
+
+		$query_params['context']['default'] = 'view';
+
+		$query_params['channel'] = array(
+			'default'     => array(),
+			'description' => __( 'Limit result set to notifications assigned to specific channel IDs.' ),
+			'type'        => 'array',
+			'items'       => array(
+				'type' => 'integer',
+			),
+		);
+
+		$query_params['offset'] = array(
+			'description' => __( 'Offset the result set by a specific number of items.' ),
+			'type'        => 'integer',
+		);
+
+		$query_params['order'] = array(
+			'description' => __( 'Order sort attribute ascending or descending. Requires authorization.' ),
+			'type'        => 'string',
+			'default'     => 'desc',
+			'enum'        => array(
+				'asc',
+				'desc',
+			),
+		);
+
+		$query_params['orderby'] = array(
+			'description' => __( 'Sort collection by notifications attribute. Requires authorization.' ),
+			'type'        => 'string',
+			'default'     => 'created_at',
+			'enum'        => array(
+				'created_at',
+				'dismissed_at',
+				'displayed_at',
+			),
+		);
+
+		$query_params['status'] = array(
+			'description' => __( 'Limit result set to notifications assigned a specific status.' ),
+			'default'     => 'undisplayed',
+			'type'        => 'string',
+			'enum'        => array(
+				'undisplayed',
+				'displayed',
+				'dismissed',
+			),
+		);
+
+		return $query_params;
+	}
```

Comment:
These were just some that I thought might be helpful, please help with determining what would be best.

Reply:
The only one that seems missing (to me) is the "count" to limit the number of queried objects

Reply:
That is handled by the WP_REST_Controller that this class inherits from, and it's called "page".

Reply:
In this case I would say everything is there, and if something is missing, we can add it as we go!

@Sephsekla what do you think? Do you think the possible parameters are all there, at least the main ones for an MVP?

Reply:
@erikyo, the page parameters are handled by the parent rest controller. https://github.com/WordPress/wordpress-develop/blob/trunk/src/wp-includes/rest-api/endpoints/class-wp-rest-controller.php

Reply:
This looks good to me, the more we can just use from the base controller, the better. We can always extend it, as well.